src/services/QueueService.ts

```diff
@@ -1,17 +1,23 @@
 import { DateTime } from "luxon";
+import ActiveMatchRepository from "../repositories/ActiveMatchRepository";
 import QueueRepository from "../repositories/QueueRepository";
 import { AddBallChaserToQueueInput, PlayerInQueue } from "../repositories/QueueRepository/types";
 
 export async function joinQueue(userId: string, userName: string): Promise<ReadonlyArray<PlayerInQueue>> {
   const queueMember = await QueueRepository.getBallChaserInQueue(userId);
 
-  if (!queueMember) {
+  //Check if in Active Match so as not to add to the queue.
+  const activeMatchMember = await ActiveMatchRepository.isPlayerInActiveMatch(userId);
```

Comment:
To help simplify some of the if logic, what if we moved this above the `getBallChaserInQueue` call and then do an early return if the player is in an active match? That will skip any unnecessary async calls and simplify some of the conditional logic later on. Thoughts?

Reply:
Sounds good.  I moved it to the top.  

Makes sense to not have a big if else chain.

Reply:
I don't think we need to call `getAllBallChasersInQueue` if the member is in the active match since nothing is expected to change. If the action is invalid it should be more or less ignored.

Maybe return null or throw an error to be handled in the controller so that it knows not to do anything.

Reply:
I am going to return null and then just null check in Interactions.ts.  Will be fixed in next commit.